daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/transaction/PartialTransaction.scala

```diff
@@ -333,6 +343,28 @@ private[lf] case class PartialTransaction(
         IncompleteTransaction(this)
     }
 
+  // construct an IncompleteTransaction from the partial-transaction
+  def finishIncomplete: TxIncompleteTransaction = {
+
+    @tailrec
+    def unwindToExercise(
+        contextInfo: PartialTransaction.ContextInfo
+    ): Option[PartialTransaction.ExercisesContextInfo] = contextInfo match {
+      case ctx: PartialTransaction.ExercisesContextInfo => Some(ctx)
+      case ctx: PartialTransaction.TryContextInfo =>
+        unwindToExercise(ctx.parent.info)
+      case _: PartialTransaction.RootContextInfo => None
+    }
+
+    ConcreteIncompleteTx(
+      GenTransaction(
+        nodes,
+        ImmArray(context.children.toImmArray.toSeq.sortBy(_.index)),
```

Comment:
oh I see here is the bug @remyhaemmerle-da mentioned: This should be the root nodes of the partial transaction not the children of the context.

Reply:
We still need to "unwind" completely the partial transaction.